src/EFCore.PG/Query/ExpressionVisitors/NpgsqlSqlTranslatingExpressionVisitor.cs

```diff
@@ -20,39 +18,15 @@ namespace Npgsql.EntityFrameworkCore.PostgreSQL.Query.ExpressionVisitors
     public class NpgsqlSqlTranslatingExpressionVisitor : SqlTranslatingExpressionVisitor
     {
         /// <summary>
-        /// The <see cref="MethodInfo"/> for <see cref="DbFunctionsExtensions.Like(DbFunctions,string,string)"/>.
+        /// The current query model visitor.
         /// </summary>
-        [NotNull] static readonly MethodInfo Like2MethodInfo =
-            typeof(DbFunctionsExtensions)
-                .GetRuntimeMethod(nameof(DbFunctionsExtensions.Like), new[] { typeof(DbFunctions), typeof(string), typeof(string) });
-
-        /// <summary>
-        /// The <see cref="MethodInfo"/> for <see cref="DbFunctionsExtensions.Like(DbFunctions,string,string, string)"/>.
-        /// </summary>
-        [NotNull] static readonly MethodInfo Like3MethodInfo =
-            typeof(DbFunctionsExtensions)
-                .GetRuntimeMethod(nameof(DbFunctionsExtensions.Like), new[] { typeof(DbFunctions), typeof(string), typeof(string), typeof(string) });
-
-        // ReSharper disable once InconsistentNaming
-        /// <summary>
-        /// The <see cref="MethodInfo"/> for <see cref="NpgsqlDbFunctionsExtensions.ILike(DbFunctions,string,string)"/>.
-        /// </summary>
-        [NotNull] static readonly MethodInfo ILike2MethodInfo =
-            typeof(NpgsqlDbFunctionsExtensions)
-                .GetRuntimeMethod(nameof(NpgsqlDbFunctionsExtensions.ILike), new[] { typeof(DbFunctions), typeof(string), typeof(string) });
-
-        // ReSharper disable once InconsistentNaming
-        /// <summary>
-        /// The <see cref="MethodInfo"/> for <see cref="NpgsqlDbFunctionsExtensions.ILike(DbFunctions,string,string,string)"/>.
-        /// </summary>
-        [NotNull] static readonly MethodInfo ILike3MethodInfo =
-            typeof(NpgsqlDbFunctionsExtensions)
-                .GetRuntimeMethod(nameof(NpgsqlDbFunctionsExtensions.ILike), new[] { typeof(DbFunctions), typeof(string), typeof(string), typeof(string) });
+        [NotNull] readonly RelationalQueryModelVisitor _queryModelVisitor;
 
         /// <summary>
-        /// The query model visitor.
+        /// The current query compilation context.
         /// </summary>
-        [NotNull] readonly RelationalQueryModelVisitor _queryModelVisitor;
+        [NotNull]
+        RelationalQueryCompilationContext Context => _queryModelVisitor.QueryCompilationContext;
```

Comment:
Probably a good idea to maintain the full name (`QueryCompilationContext`) since `Context` is quite ambiguous within EF Core (e.g. `DbContext`).